R/tag_query.R

```diff
@@ -306,7 +303,31 @@ tagEnvToTags_ <- function(x) {
     # undo parent env and key
     x$parent <- NULL
     x$envKey <- NULL
-    # recurse through children
+
+    # Reorder the names to match a typical tag() structure that has `name`, `attribs`, and `children` first
+    # Avoid calling `setdiff()` if possible (it is slow).
+    if (!is.character(x[["name"]])) {
+      stop("A tag environment has lost its `$name`. Did you remove it?")
+    }
+    if (is.null(x[["attribs"]])) x$attribs <- setNames(list(), character(0))
+    if (is.null(x[["children"]])) x$children <- list()
+
+    xNames <- names(x)
+    newNames <- c(
+      "name", "attribs", "children",
+      if (length(xNames) > 3) {
+        setdiff(xNames, c("name", "attribs", "children"))
+      }
+    )
+    if (!identical(xNames, newNames)) {
+      # Need to preserve attributes. Must move values in two steps.
+      # Reorder values
+      x[] <- x[newNames]
+      # Reorder names
+      names(x) <- newNames
```

Comment:
I don't think this line is necessary, is it?

Reply:
To keep the ordering consistent with `tag()`, it is necessary. (It is not necessary with `mget()`, which is really fast.)

Reply:
("Resolving" as using `mget()` approach, which removes this code)

Reply:
I know it's moot now, but:

```R
x <- list(a=1, b=2)
x[c("b", "a")]
#> $b
#> [1] 2
#> 
#> $a
#> [1] 1
```

Reply:
This drops attributes

```r
x <- list(a = 1, b =2)
attr(x, "hello") <- "world"

str(x)
#> List of 2
#>  $ a: num 1
#>  $ b: num 2
#>  - attr(*, "hello")= chr "world"

str(x[c("b", "a")])
#> List of 2
#>  $ b: num 2
#>  $ a: num 1
```